include/gsl/gsl_util

```diff
@@ -43,6 +43,12 @@
 #define GSL_CONSTEXPR_NARROW 1
 #endif
 
+#if defined(__cplusplus) && (__cplusplus >= 201703L)
```

Comment:
This should have the form
```C++
#ifdef __has_cpp_attribute                      // Check if __has_cpp_attribute is present
#  if __has_cpp_attribute(deprecated)           // Check for an attribute
#    define DEPRECATED(msg) [[deprecated(msg)]]
#  endif
#endif
#ifndef DEPRECATED
#    define DEPRECATED(msg)
#endif
```
See https://en.cppreference.com/w/cpp/feature_test.

Reply:
I wonder if it's even necessary. GSL targets c++14 and up. Attributes were standardized in c++11. Wouldn't it make sense to use [[nodiscard]] directly?

Reply:
Neither form is foolproof. The current check isn't enough to suggest the existence of `[[nodiscard]]` and not all compilers with `__cplusplus >= 201703L` implement the relaxation of allowing non-existent attributes in case they haven't implemented `nodiscard`. I'd guess some users that use `-Werror` would get a hard error as such. My suggestion itself requires a C++20 feature, so I can't say it's better. `nodiscard` itself is a C++17 addition, as noted in the first table of the link.

Reply:
I think `__has_cpp_attribute` is the best we can do for now. I am glad that this feature is being added in c++20. But at the same time I hate that it took so long to realize the need.